### PR TITLE
[Refactor] refactor wan2.2 diffuse && add ut

### PR DESCRIPTION
**Pull request by bjf-frz**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Purpose

This pr aims to refactor wan2.2 extrace `diffuse` function from `forward`.

## Test Result

After refactoring, `diffuse` durations hooked and the result displayed in benchmark.
```
================= Serving Benchmark Result =================
Backend:                                 v1/videos      
Model:                                   Wan2.2_I2V_A14B_diffusers
Dataset:                                 random         
Task:                                    i2v            
--------------------------------------------------
Benchmark duration (s):                  12.17          
Request rate:                            inf            
Max request concurrency:                 1              
Successful requests:                     1/1              
--------------------------------------------------
Request throughput (req/s):              0.08           
Latency Mean (s):                        12.1732        
Latency Median (s):                      12.1732        
Latency P99 (s):                         12.1732        
Latency P95 (s):                         12.1732        
--------------------------------------------------
Peak Memory Max (MB):                    74178.00       
Peak Memory Mean (MB):                   74178.00       
Peak Memory Median (MB):                 74178.00       
--------------------------------------------------
Stage Durations Mean (s):
  Wan22I2VPipeline.vae.encode:           0.9670         
  Wan22I2VPipeline.scheduler_step_maybe_with_cfg: 0.0038         
  Wan22I2VPipeline.diffuse:              5.1517         
  Wan22I2VPipeline.vae.decode:           1.5907         

============================================================
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan. Please provide the test scripts & test commands. Please state the reasons if your codes don't require additional test scripts. For test file guidelines, please check the [test style doc](https://docs.vllm.ai/projects/vllm-omni/en/latest/contributing/ci/tests_style/)
- [ ] The test results. Please paste the results comparison before and after, or the e2e results.
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model. **Please run `mkdocs serve` to sync the documentation editions to `./docs`.**
- [ ] (Optional) Release notes update. If your change is user-facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Review by SamitHuang — APPROVED** [review]
The extraction of the `diffuse` method preserves the timestep expansion and CFG logic accurately. The unit tests in `test_wan22_pipeline_diffuse.py` correctly isolate the denoising loop and verify scheduler interactions. Please ensure the PR description checklist is completed with the test plan and results before merging.

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py [hunk 1]
```diff
@@ -571,90 +670,19 @@ def forward(
 
         if DEBUG_PERF:
```
Comment (hsliuustc0106) [inline vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py:726]: what is DEBUG_PERF?

**Review by SamitHuang — CHANGES_REQUESTED** [review]
pls fix the linting error

**Comment by bjf-frz** [comment]
> pls fix the linting error

thx, fixed

**Comment by hsliuustc0106** [comment]
Clean refactor. Good test coverage with `test_forward_delegates_denoising_to_diffuse` and `test_diffuse_runs_prediction_and_scheduler_for_each_timestep`.

**Review by gcanlin — APPROVED** [review]
LGTM, just make ensure not to change to the logic in `diffuse` func.

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py (hunk 1 above)
Comment (bjf-frz) [reply vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py:726]: seems to be some unused debug code that was left behind.

File: vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py [hunk 2]
```diff
@@ -336,6 +336,105 @@ def num_timesteps(self):
...
+                    if self.transformer is not None:
+                        current_model = self.transformer
+                    elif self.transformer_2 is not None:
+                        # Fallback to transformer_2 if transformer not loaded
+                        current_model = self.transformer_2
+                    else:
+                        raise RuntimeError("No transformer available for high-noise stage")
+
+                if self.expand_timesteps and latent_condition is not None:
+                    if first_frame_mask is None:
```
Comment (gcanlin) [inline vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py]: I notice that these two lines are newly added.

Comment (bjf-frz) [reply vllm_omni/diffusion/models/wan2_2/pipeline_wan2_2.py]: > I notice that these two lines are newly added.

All unexpected changes were rolled back.

**Comment by hsliuustc0106** [comment]
the sum of each stage is not equal to the duration, where is the gap?

**Comment by bjf-frz** [comment]
> the sum of each stage is not equal to the duration, where is the gap?

It did not include text_encoder.forward, scheduler.set_timestamp, etc.

File: tests/diffusion/models/wan2_2/test_wan22_pipeline_diffuse.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+from contextlib import contextmanager
+from types import SimpleNamespace
+
+import pytest
+import torch
+from torch import nn
+
+from vllm_omni.diffusion.models.wan2_2.pipeline_wan2_2 import Wan22Pipeline
+
+pytestmark = [pytest.mark.core_model, pytest.mark.cpu, pytest.mark.diffusion]
```
Comment (lishunyang12) [inline tests/diffusion/models/wan2_2/test_wan22_pipeline_diffuse.py:13]: Only `Wan22Pipeline.diffuse` is covered here, but this PR also adds `diffuse` overrides to `Wan22I2VPipeline`, `Wan22TI2VPipeline`, and `Wan22VACEPipeline`. Can you extend the uts to cover those as well? Otherwise the three subclass bodies will drift silently.

**Comment by david6666666** [comment]
pls fix ci

**Comment by bjf-frz** [comment]
> pls fix ci

after updated branch, now it passed the ci

File: tests/diffusion/models/wan2_2/test_wan22_pipeline_diffuse.py (hunk 3 above)
Comment (wtomin) [reply tests/diffusion/models/wan2_2/test_wan22_pipeline_diffuse.py:13]: Agree with this suggestion.

Comment (bjf-frz) [reply tests/diffusion/models/wan2_2/test_wan22_pipeline_diffuse.py:13]: thx @wtomin @lishunyang12 , relevant uts has added and passed
